src/app/routes/profile.js

```diff
@@ -1,22 +1,63 @@
 import { Hono } from 'hono'
 import { html } from 'hono/html'
-import { Layout, UserInfo } from './../../components.js'
+import { Layout, UserInfo, AddPasskeySection, PasskeyList, Message } from './../../components.js'
+import { auth } from '../../../auth.js'
 
 export default new Hono()
   .get('/profile', async (c) => { // Profile page
     const user = c.get('user');
     if (!user) return c.redirect('/login?error=no+session');
 
+    const error = c.req.query('error');
+    const success = c.req.query('success');
+
+    // Fetch user's passkeys
+    let passkeys = [];
+    try {
+      passkeys = await auth.api.listPasskeys({
+        headers: c.req.raw.headers,
+      });
+    } catch (error) {
+      console.error('Failed to fetch passkeys:', error);
+    }
+
     return c.html(Layout({
       title: 'Profile',
       children: html`
         <h1>Profile</h1>
+        
+        ${Message({ error, success })}
+        
         ${UserInfo({ user })}
         
-        <form method="post" action="/logout" style="display: inline;">
-          <button type="submit">Logout</button>
-        </form>
-        <a href="/">Back to Home</a>
+        ${PasskeyList({ passkeys })}
+        
+        ${AddPasskeySection({ user })}
+        
+        <div style="margin-top: 30px;">
```

Comment:
Do we really want inline styles? Does it have to look nice at this point?
```suggestion
        <div>
```